include/pinocchio/visualizers/base-visualizer.hpp

```diff
@@ -0,0 +1,227 @@
+      {
+      }
+
+      /// @brief Set the active camera position.
+      virtual void setCameraPosition(const Eigen::Ref<const Vector3> & /*position*/)
+      {
+      }
+
+      /// @brief Set the active camera 6D pose.
+      virtual void setCameraPose(const Eigen::Ref<const Matrix4> & /*pose*/)
+      {
+      }
+
+      /// @copybrief setCameraPose()
+      inline void setCameraPose(const SE3 & pose)
+      {
+        this->setCameraPose(pose.toHomogeneousMatrix());
+      }
+
+      /// @brief Set camera zoom level; what this means depends on the
+      /// implementation (FOV zoom or moving forwards).
+      virtual void setCameraZoom(Scalar /*value*/)
+      {
+      }
+
+      /// @brief Enable/disable controlling the camera from keyboard and mouse.
+      virtual void enableCameraControl(bool)
+      {
+      }
+
+      /// @brief Delete all objects from the scene.
+      virtual void clean()
+      {
+      }
+
+      const Model & model() const
+      {
+        return m_model;
+      }
+
+      const GeometryModel & visualModel() const
+      {
+        return *m_visualModel;
+      }
+
+      const GeometryModel & collisionModel() const
+      {
+        PINOCCHIO_THROW(
+          hasCollisionModel(), std::logic_error, "No collision model in the visualizer.");
+        return *m_collisionModel;
+      }
+
+      bool hasCollisionModel() const
+      {
+        return m_collisionModel != nullptr;
+      }
+
+      /// @brief Whether the internal data pointers are borrowed (external), or owned.
+      bool hasExternalData() const
+      {
+        return !m_ownedData;
+      }
+
+      Data & data()
+      {
+        return *m_data;
+      }
+      const Data & data() const
+      {
+        return *m_data;
+      }
+
+      GeometryData & visualData()
+      {
+        return *m_visualData;
+      }
+      const GeometryData & visualData() const
+      {
+        return *m_visualData;
+      }
+
+      GeometryData & collisionData()
+      {
+        PINOCCHIO_THROW(
+          hasCollisionModel(), std::logic_error, "No collision model in the visualizer.");
+        return *m_collisionData;
+      }
+
+      const GeometryData & collisionData() const
+      {
+        PINOCCHIO_THROW(
+          hasCollisionModel(), std::logic_error, "No collision model in the visualizer.");
+        return *m_collisionData;
+      }
+
+    protected:
+      std::reference_wrapper<Model const> m_model;
+      GeometryModel const * m_visualModel;
+      GeometryModel const * m_collisionModel;
+
+      Data * m_data;
+      GeometryData * m_visualData;
+      GeometryData * m_collisionData;
+      bool m_ownedData;
+
+      /// @brief This method is called at the beginning of display().
+      virtual void displayPrecall()
+      {
+      }
+      virtual void displayImpl() = 0;
+
+      void destroyData();
+    };
+  } // namespace visualizers
+} // namespace pinocchio
+
+namespace pinviz = ::pinocchio::visualizers; // NOLINT
```

Comment:
I don't if it is a good idea to add it. If it is not used, maybe we should let the user define it in its own project I would say.

Reply:
You're right, this was a holdover from the upstream `pinocchio-visualizers`.

Reply:
```suggestion
```